code/data_structures/linked_list/doubly_linked_list/doubly_linked_list.js

```diff
@@ -0,0 +1,516 @@
+            node.next = current_node;
+
+            current_node.prev.next = node;
+
+            current_node.prev = node;
+        }
+
+        this._length += 1;
+    }
+};
+
+/**
+ * Inserts the element at the begining of the list
+ * @param {any} element - The element to be added
+ */
+LinkedList.prototype.addFirst = function (element) {
+    this.add(element, 0);
+};
+
+/**
+ * Inserts the element at the end of the list
+ * @param {any} element - The element to be added
+ */
+LinkedList.prototype.addLast = function (element) {
+    this.add(element);
+};
+
+/**
+ * Removes all the elements of the list
+ */
+LinkedList.prototype.clear = function () {
+    while (this._length) {
+        this.pop();
+    }
+};
+
+/**
+ * Returns if the list contains the element
+ * @param {any} element - The element to be checked
+ * @returns {boolean} True if the list contains the element
+ */
+LinkedList.prototype.contains = function (element) {
+    var found_element = false;
+
+    this.every(function (elem, index) {
+        if (elem === element) {
+            found_element = true;
+            return false;
+        }
+        return true;
+    });
+
+    return found_element;
+};
+
+/**
+ * The function called by the forEach method.
+ * @callback LinkedList~iterateCallback
+ * @function
+ * @param {any} element - The current element
+ * @param {number} [index] - The index of the current element
+ * @param {LinkedList} [list] - The linked list it is using
+ */
+
+/**
+ * Performs the fn function for all the elements of the list, untill the callback function returns false
+ * @param {iterateCallback} fn - The function the be executed on each element
+ * @param {object} [this_arg] - The object to use as this when calling the fn function 
+ */
+LinkedList.prototype.every = function (fn, this_arg) {
+    var current_node = this._front;
+    var index = 0;
+
+    while (current_node) {
+        if (!fn.call(this_arg, current_node.value, index, this)) {
+            break;
+        }
+        current_node = current_node.next;
+        index += 1;
+    }
+};
+
+/**
+ * Performs the fn function for all the elements of the list
+ * @param {iterateCallback} fn - The function the be executed on each element
+ * @param {object} [this_arg] - The object to use as this when calling the fn function
+ */
+LinkedList.prototype.forEach = function (fn, this_arg) {
+    var current_node = this._front;
+    var index = 0;
+
+    while (current_node) {
+        fn.call(this_arg, current_node.value, index, this);
+        current_node = current_node.next;
+        index += 1;
+    }
+};
+
+/**
+ * Returns the element at the index, if any
+ * @param {number} index - The index of the element to be returned
+ * @returns {any} The element at the specified index, if any. Returns undefined otherwise
+ */
+LinkedList.prototype.get = function (index) {
+
+    if (index < 0) {
+        return undefined;
+    }
+
+    var element;
+
+    this.every(function (elem, i) {
```

Comment:
A for loop would be clarify the intent more of this function (`get`), and shorten it.